SalesforceSDKCore.podspec

```diff
@@ -25,9 +25,8 @@ Pod::Spec.new do |s|
       sdkcore.resource = 'shared/resources/SalesforceSDKAssets.xcassets'
 
       sdkcore.subspec 'base' do |sp|
```

Comment:
Maybe we should add a comment about what goes in `base`, what goes in `arc`, etc., so we remember next time around? Basically, the description on this PR.